include/seastar/core/sharded.hh

```diff
@@ -1029,6 +1035,12 @@ public:
         _cpu = other._cpu;
         return *this;
     }
+    /// Return the use_count of the wrapped shared objects
+    template <typename U = PtrType>
+    std::enable_if_t<HasUseCount<U>, long>
+    use_count() const noexcept {
+        return _value.use_count();
```

Comment:
How can this work? The use count can be incremented remotely on the owning shard.

Reply:
yes, something like that, just wit clear_gently.

> How can this work? The use count can be incremented remotely on the owning shard.

We can throw if it's not called on the owner shard and document that requirement.

Alternatively, we can expose a method to return a (const) reference to the contained smart/shared pointer that needs to be called also only on the owner shard, and let the caller call use_count and the shared pointer.